### PR TITLE
feat(duckdb): use `read_json_auto` when reading json

### PR DESCRIPTION
**Pull request by cpcloud**
This PR changes the duckdb read_json implementation to use the new-in-duckdb-0.7.0 function `read_json_auto`. Depends on #5545.

### CODE REVIEW

File: ibis/backends/duckdb/__init__.py [hunk 1]
```diff
@@ -232,14 +233,16 @@ def read_json(
     ) -> ir.Table:
         """Read newline-delimited JSON into an ibis table.
 
+        !!! note "This feature requires duckdb>=0.7.0"
```
Comment (jcrist) [inline ibis/backends/duckdb/__init__.py:236]: Is there a way we could check that this is true on the connection, and raise a better error message if the `duckdb` version is too low? If this is hard to do don't worry about it.

File: ibis/expr/api.py [hunk 2]
```diff
@@ -880,7 +880,27 @@ def read_json(sources: str | Path | Sequence[str | Path], **kwargs: Any) -> ir.T
...
+    >>> import json
+    >>> from pathlib import Path
+    >>> import ibis
+    >>> ibis.options.interactive = True
+    >>> rows = [
+    ...    {"a": 1, "b": "d"},
+    ...    {"a": 2, "b": None},
+    ...    {"a": None, "b": "f"},
+    ... ]
+    >>> Path("lines.json").write_text("\n".join(map(json.dumps, rows)))
```
Comment (jcrist) [inline ibis/expr/api.py:892]: Simplifying the example to reduce lines before the user sees Ibis code:

```suggestion
    >>> import ibis
    >>> ibis.options.interactive = True
    >>> jsonlines = """
    ... {"a": 1, "b": "d"}
    ... {"a": 2, "b": null}
    ... {"a": null, "b": "f"}
    ... """
    >>> open("lines.json", "w").write(jsonlines)
```

File: ibis/backends/duckdb/__init__.py (hunk 1 above)
Comment (cpcloud) [reply ibis/backends/duckdb/__init__.py:236]: Yep, lemme give it a shot.

Comment (cpcloud) [reply ibis/backends/duckdb/__init__.py:236]: Fixed. Let me know if the error message is sufficient.

**Review by jcrist — APPROVED** [review]

**Pull request merged**